src/node_modules/app/elasticsearch/types/genericTypes.js

```diff
@@ -123,12 +154,15 @@ export const constant = ({type, shape, value}) => ({
 	...shape ? {[shapeKey]: shape} : {},
 });
 
-// numString (parses as a number)
-export const numString = _shape('numString');
-export const numStringOf = stringValue => ({
-	...numString,
-	stringValue,
+// intWithUnit
+export const intWithUnitKey = 'intWithUnit';
+export const intWithUnitD = _shape(intWithUnitKey);
+export const intWithUnitOf = (units, defaultUnit) => ({
+	...intWithUnitD(),
+	__value: integerD(1),
+	__unit: enumsOf(units, defaultUnit)
```

Comment:
no need to use `__` here

Reply:
The `__` in this case helps denote internal keys/props/params to the form generator, which allows it to know the field names should not be visible in the generated form. We can remove them but we'd have to add more logic to hide the field labels when generating the form.

Reply:
The logic should be the same as for other types (e.g. `extentOf` has `min`, `max`, not `__min` and `__max`): how we use the params used to describe a type should not the type business, so I'd still remove `__` and move that logic where it belongs, that is in the form generator.